### ISSUE TITLE
Run `make tre-stop` at the end of the TRE Deploy github workflow

### ISSUE THREAD

**Issue by marrobi** [microsoft/AzureTRE#1115]
## Description

As a Azure Administrator
I want to shut down/deallocate resources at the end of a CD run
So that Azure spend is minimized

## Acceptance criteria

- [x] After e2e tests run - pass or fail - run `make tre-stop`
- [x] Ensure that all resources are started prior to e2e tests running.

**Comment by marrobi** [comment]
I tried this, but need to handle az logon. Might be worth getting #1132 sorted first.
 
 ```
  tre_stop:
      name: "Deallocate Resources"
      runs-on: ubuntu-latest
      environment: Dev
      needs: [e2e_tests]
      if: |
        ${{
        always() &&
        github.event.name == 'schedule'
        || github.event_name == 'push'
        || github.event_name == 'workflow_dispatch'
        || contains(github.event.pull_request.labels.*.name, 'safe to test')
        }}
      steps:
        - name: Checkout code
          uses: actions/checkout@v2
        - name: Azure Login
          uses: azure/login@v1
          with:
            creds: ${{ secrets.AZURE_CREDENTIALS }}
        - name: Deallocate Resources
          shell: bash
          env:
            TRE_ID: "${{secrets.TRE_ID}}"
          run: |
            export ARM_SUBSCRIPTION_ID=$(echo "$AZURE_CREDENTIALS" | jq -r '.subscriptionId')
            pwsh -c Install-Module Az -Force -Scope CurrentUser
            make tre-stop
```

**Comment by marrobi** [comment]
Fixed in #973
